bundles/org.eclipse.equinox.common.tests/src/org/eclipse/equinox/common/tests/ServiceCallerTest.java

```diff
@@ -209,22 +205,20 @@ public void testCall() throws IOException, BundleException, ClassNotFoundExcepti
 		}
 	}
 
+	@Test
 	public void testInvalidFilter() {
-		try {
-			new ServiceCaller<>(getClass(), IServiceExample.class, "invalid filter");
-			fail("Expected an exception on invalid filter.");
-		} catch (IllegalArgumentException e) {
-			assertTrue("Unexpected cause.", e.getCause() instanceof InvalidSyntaxException);
-		}
-		try {
+		IllegalArgumentException exceptionOnConstructor = assertThrows(IllegalArgumentException.class,
+				() -> new ServiceCaller<>(getClass(), IServiceExample.class, "invalid filter"));
+		assertTrue("Unexpected cause.", exceptionOnConstructor.getCause() instanceof InvalidSyntaxException);
```

Comment:
This could be integrated into `assertThrows()`.

Reply:
Can you explain how you would expect this should look like? Since this assertions validates the type of the cause of the exception and not the type of the exception itself, I am not sure how I could integrate that validation into the `assertThrows`. I have at least replaced the simple `assertTrue` with an instanceof matcher to get a better message on failure in 53398de.

Reply:
Sorry I oversaw that detail (I did the review on my phone and obviously my screen isn't big enough 😅).
It is fine as it is.